azure/services/scalesets/scalesets.go

```diff
@@ -673,27 +578,57 @@ func (s *Service) getVirtualMachineScaleSetDefaultNetworkConfiguration(vmssSpec
 				})
 		}
 	}
-	return &[]compute.VirtualMachineScaleSetNetworkConfiguration{{
-		Name: pointer.String(vmssSpec.Name),
-		VirtualMachineScaleSetNetworkConfigurationProperties: &compute.VirtualMachineScaleSetNetworkConfigurationProperties{
-			Primary:            pointer.Bool(true),
-			EnableIPForwarding: pointer.Bool(true),
-			IPConfigurations: &[]compute.VirtualMachineScaleSetIPConfiguration{
-				{
-					Name: pointer.String(vmssSpec.Name),
-					VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{
-						Subnet: &compute.APIEntityReference{
-							ID: pointer.String(azure.SubnetID(s.Scope.SubscriptionID(), vmssSpec.VNetResourceGroup, vmssSpec.VNetName, vmssSpec.SubnetName)),
-						},
-						Primary:                         pointer.Bool(true),
-						PrivateIPAddressVersion:         compute.IPVersionIPv4,
-						LoadBalancerBackendAddressPools: &backendAddressPools,
+	nicConfigs := []compute.VirtualMachineScaleSetNetworkConfiguration{}
+	for i, n := range vmssSpec.NetworkInterfaces {
+		nicConfig := compute.VirtualMachineScaleSetNetworkConfiguration{}
+		nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties = &compute.VirtualMachineScaleSetNetworkConfigurationProperties{}
+		nicConfig.Name = pointer.String(vmssSpec.Name + "-nic-" + strconv.Itoa(i))
```

Comment:
```suggestion
		nicConfig.Name = pointer.String(vmssSpec.Name + "-nic-" + strconv.Itoa(i))
		nicConfig.EnableIPForwarding = pointer.Bool(true)
```

EnableIPForwarding is necessary for the CNI to work

Reply:
Added to maintain what's there, however I'm still not convinced this should be hardcoded. For VMs it's configurable: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/networkinterfaces/spec.go#L211

Can you expand on "is necessary for the CNI to work"? Which CNI? What IP family?

Reply:
From the comment [here](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/azuremachine_types.go#L103), it's only needed for Calico + Ipv6

Reply:
I'm not the expert in CNIs but I believe it is needed for Calico with VXLAN (which is required when you use windows nodes and Calico CNI). At least not having IP Forwarding enabled was the root cause of our issues when we ugpraded CAPZ. 

Adding this flag as a property on the networkInterfaces object would be nice, however if it's not enabled it might be difficult to figure out whats wrong for a user.